test/runtests.jl

```diff
@@ -11,8 +11,36 @@ using FilePathsBase.TestPaths
 using UUIDs: uuid4
 using JSON3
 
-aws = AWSConfig()
+if VERSION >= v"1.5"
+    using Minio
+    AWS.aws_account_number(::Minio.MinioConfig) = "123"
 
+    # We run most tests under Minio. This can be done locally by those
+    # without access to the s3 bucket under which CI is performed.
+    # We then run all tests with s3 directly.
+
+    # We use multiple directories so that Minio can support versioning.
+    dirs = [mktempdir() for _ =1:12]
+    port = 9005
+    minio_server = Minio.Server(dirs; address="localhost:$port")
+
+    # We use this boolean flag to skip some tests under Minio
+    minio = true
```

Comment:
Remind me why we have this global instead of checking the `global_aws_config` type?

Reply:
just because we use it in lots of places. We could do `!(aws isa AWSConfig)` instead in these local places. (We cannot do `aws isa MinioConfig` since Minio may not be loaded)

Reply:
Ah, good point about Minio not being loaded. I'd probably use a function to encapsulate any logic needed there. Something you could do is define `abstract type MinioConfig end` in cases where Minio.jl isn't loaded